app/user.go

```diff
@@ -629,12 +629,13 @@ func (a *App) GetChannelGroupUsers(channelID string) ([]*model.User, *model.AppE
 	return result.Data.([]*model.User), nil
 }
 
-func (a *App) GetUsersByIds(userIds []string, asAdmin bool, viewRestrictions *model.ViewUsersRestrictions) ([]*model.User, *model.AppError) {
-	result := <-a.Srv.Store.User().GetProfileByIds(userIds, viewRestrictions == nil, viewRestrictions)
+func (a *App) GetUsersByIds(userIds []string, options *model.UserGetByIdsOptions) ([]*model.User, *model.AppError) {
+	result := <-a.Srv.Store.User().GetProfileByIds(userIds, options, options.ViewRestrictions == nil)
```

Comment:
might be nice to pre-declare `allowFromCache := options.ViewRestrictions == nil` to help document what's going on.

Reply:
Makes sense